### ISSUE TITLE
Include more packages in separate requirements files

### ISSUE THREAD

**Issue by mdpiper** [csdms/bmi-example-python#24]
Should we expand the set of requirements files for this repository? I'm thinking about four separate files.

**requirements.txt** -- packages needed to install with `pip install -e .`
- numpy
- scipy
- pyyaml
- bmipy

**requirements-dev.txt** -- developer packages used with the repo Makefile
- make
- build
- twine
- flake8
- black
- isort

**requirements-testing.txt** -- packages used for running tests
- coveralls
- pytest
- pytest-cov
- six
- bmi-tester

**requirements-examples.txt** -- packages used for running examples
- notebook

**Comment by mcflugen** [comment]
Yes, but they should also go in the *pyproject.toml* file as something like,
```toml
[project.optional-dependencies]
dev = [
  "black",
  "flake8",
  "isort",
]
testing = [
  "coveralls",
  "pytest",
  "bmi-tester"
]
examples = [
  "notebook",
]
```
This allows users to *pip* install like, `pip install -e ".[dev]"`. The requirements files are really only there for *mamba* users who want to do `mamba install --file=requirements-testing.txt` but having requirements in two locations isn't great. For other projects, I include a *requirements.py* script that extracts requirements from *pyproject.toml* that can be use to create the various requirements files.

I'm not totally sure what to do with dependencies like *make* that aren't *pip*-installable or available on all platforms. In the case of *make*, I think I would lean toward just dropping it as a dependency. If someone wants to use the *Makefile*, they should know they need to install *make*.

*six* should not be a dependency. We should remove it in a separate pull request.

**Comment by mdpiper** [comment]
This is super helpful, thank you!

**Comment by mdpiper** [comment]
As a note to myself, a recent example of removing *six*: https://github.com/csdms/pymt/pull/159.
